### PR TITLE
Add Kafka driver

### PR DESCRIPTION
**Pull request by tsecheran**

### CODE REVIEW

File: pifpaf/drivers/kafka.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import os
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: missing headers

File: pifpaf/drivers/kafka.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from pifpaf import drivers
+
+
+class KafkaDriver(drivers.Driver):
+    DEFAULT_PORT = 9092
+
+    def __init__(self, port=DEFAULT_PORT, **kwargs):
+        super(KafkaDriver, self).__init__(**kwargs)
+        self.port = port
+        self.path = ["/opt/kafka/bin"]
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: no need to init that in each instance, go ahead with a class var at this stage

File: pifpaf/drivers/kafka.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    def _setUp(self):
+        super(KafkaDriver, self)._setUp()
+        c, _ = self._exec('kafka-server-start.sh', '-daemon',
+                          "server.properties",
+                          "--override",
+                          "port="+str(self.port), path=self.path)
+
+        self.putenv("KAFKA_PORT", self.port)
+        self.putenv("KAFKA_URL", "localhost:%d" % self.port)
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: you miss kafka://

File: pifpaf/drivers/kafka.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        super(KafkaDriver, self).__init__(**kwargs)
+        self.port = port
+        self.path = ["/opt/kafka/bin"]
+
+    def _setUp(self):
+        super(KafkaDriver, self)._setUp()
+        c, _ = self._exec('kafka-server-start.sh', '-daemon',
+                          "server.properties",
+                          "--override",
+                          "port="+str(self.port), path=self.path)
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: This is never killed?

File: pifpaf/drivers/kafka.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+        self.putenv("KAFKA_PORT", self.port)
+        self.putenv("KAFKA_URL", "localhost:%d" % self.port)
+
+    def start_kafka(self):
+        c, _ = self._exec('kafka-server-start.sh', '-daemon', path=self.path)
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: return values are unused?

File: pifpaf/drivers/kafka.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        super(KafkaDriver, self)._setUp()
+        c, _ = self._exec('kafka-server-start.sh', '-daemon',
+                          "server.properties",
+                          "--override",
+                          "port="+str(self.port), path=self.path)
+
+        self.putenv("KAFKA_PORT", self.port)
+        self.putenv("KAFKA_URL", "localhost:%d" % self.port)
+
+    def start_kafka(self):
```
Comment (jd) [inline pifpaf/drivers/kafka.py]: maybe you could use that method in _setUp() ?

File: pifpaf/drivers/kafka.py (hunk 1 above)
Comment (tsecheran) [inline pifpaf/drivers/kafka.py]: Will fix this.

File: pifpaf/drivers/kafka.py (hunk 2 above)
Comment (tsecheran) [inline pifpaf/drivers/kafka.py]: Ok changed that...

File: pifpaf/drivers/kafka.py (hunk 3 above)
Comment (tsecheran) [inline pifpaf/drivers/kafka.py]: Added that, thanks

File: pifpaf/drivers/kafka.py (hunk 6 above)
Comment (tsecheran) [inline pifpaf/drivers/kafka.py]: Yes, changed this

File: pifpaf/drivers/kafka.py (hunk 5 above)
Comment (tsecheran) [inline pifpaf/drivers/kafka.py]: Ok I will check for the return code, is there anything do I need to add here?

**Comment by jd** [comment]
Implemented by @sileht

**Pull request closed without merging**